ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/BackupReplayFile.java

```diff
@@ -84,6 +84,15 @@ public void write(final ConsumerRecord<byte[], byte[]> record) throws IOExceptio
       throw new IOException("Write permission denied.");
     }
 
+    if (record == null) {
+      throw new IllegalArgumentException("Record is null");
+    }
+
+    if (record.key() == null || record.value() == null) {
+      throw new IllegalArgumentException(String.format(
+          "Record contain a null key/value: key=%s, value=%s", record.key(), record.value()));
```

Comment:
it might also help to output the partition/offset (even though partition is always 0 here) - ditto for other error messages

Reply:
Done